openwisp_firmware_upgrader/api/serializers.py

```diff
@@ -26,6 +26,13 @@ class BaseSerializer(FilterSerializerByOrgManaged, ValidatedModelSerializer):
 
 
 class CategorySerializer(BaseSerializer):
+    def validate_organization(self, value):
+        if not value and not self.context.get('request').user.is_superuser:
+            raise serializers.ValidationError(
+                'Only superusers can create or edit shared categories'
+            )
+        return value
```

Comment:
@youhaveme9 why do we need this change, if needed can you add a testcase as well?

Reply:
I am also wondering why this is needed?

Reply:
A test was failing after pandafy added 'http': get_asgi_application(), in routing.py

```
======================================================================
FAIL: test_shared_category (openwisp_firmware_upgrader.tests.test_api.TestCategoryViews) [Test org admin cannot create shared category]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/openwisp-firmware-upgrader/openwisp-firmware-upgrader/openwisp_firmware_upgrader/tests/test_api.py", line 487, in test_shared_category
    self.assertEqual(response.status_code, 400)
AssertionError: 201 != 400
```

Reply:
https://github.com/openwisp/openwisp-firmware-upgrader/actions/runs/14411400447/job/40419939624

Reply:
That's a good catch, I am really puzzled on why that test was not failing before (it was added in https://github.com/openwisp/openwisp-firmware-upgrader/pull/293), @pandafy we'll need to look into this while working on https://github.com/openwisp/openwisp-users/issues/238, I also created an issue in this module to go back to this later on: https://github.com/openwisp/openwisp-firmware-upgrader/issues/300.

Reply:
After some debugging, I have come to conclusion that this test started to fail after upgrading [DRF in openwisp-utils](https://github.com/openwisp/openwisp-utils/commit/0a7d6e2c6de43a94fd7b938a316dd58ee27af0d0)